src/common/appId.js

```diff
@@ -1,21 +1,19 @@
 import { parseQueryString } from 'binary-common-utils/lib/tools';
-import { addTokenIfValid } from 'binary-common-utils/lib/account';
-import { set as setStorage } from 'binary-common-utils/lib/storageManager';
-
-export const setAppId = () => {
-    let appId = 0;
-    if (document.location.port === '8080') {
-        appId = 1168; // binary bot on localhost
-    } else if (document.location.hostname.indexOf('github.io') >= 0) {
-        appId = 1180; // binary bot github.io
-    } else if (document.location.pathname.indexOf('/beta') >= 0) {
-        appId = 1261; // binary bot beta
-    } else if (document.location.pathname.indexOf('/translation') >= 0) {
-        appId = 1412; // binary bot translation
-    } else {
-        appId = 1169; // binary bot
-    }
-    setStorage('appId', appId);
+import { LiveApi } from 'binary-live-api';
+import {
+    addToken,
+    removeToken,
+    getTokenList,
+    removeAllTokens,
+    get as getStorage,
+    set as setStorage,
+} from 'binary-common-utils/lib/storageManager';
+import { getLanguage } from './lang';
+
+export const setDefaultAppId = () => {
```

Comment:
Regarding these sets of functions. We can use helper functions to make sure we keep the code clean and readable

Ideally, we would like to define functions that do only one thing at a time, and they have a name that correctly describes their task.

This way we can compose more complex logic using this smaller functions, that eventually `describe` the behavior we expect, rather than `doing` what we want.

This technique (aka. [functional programming](https://opensource.com/article/17/6/functional-javascript#function-composition) has multiple benefits, such as:

- Makes the code self-explanatory
- Makes testing easier
- Makes the code more reusable (because smaller tasks may be usable elsewhere)

Here's how I would refactor the code to be more functional:

```
// Get the custom endpoint if set, ow. get the default endpoint
export function getApiEndpoint() {
    if (!hasCustomEndpoint()) return getDefaultEndpoint();

    const apiUrl = getCustomApiUrl();
    const appId = getCustomAppId();

    return {
        apiUrl,
        appId,
    };
}

// Used to show the default endpoint when custom endpoint is invalid
export function getDefaultEndpoint() {
    const apiUrl = getWebSocketUrl('frontend.binary.com');
    const appId = getDefaultAppId();

    return {
        apiUrl,
        appId,
    };
}

// get the default app id based on location
function getDefaultAppId() {
    const location = window.location.href;

    if (location.match(/github.io/)) return 1112;
    if (location.match(/beta/)) return 1113;
    if (location.match(/localhost/)) return 1114;

    return 1169;
}

// Give me the server, I'll give you the websocket URL
function getWebSocketUrl(server) {
    return `wss://${server}/websockets/v3`;
}

// Get the custom API URL From me (if any)
function getCustomApiUrl() {
    const serverUrl = getStorage('config.server_url');
    if (!serverUrl) return undefined;
    return getWebSocketUrl(serverUrl);
}

function getCustomAppId() {
    return getStorage('config.app_id');
}

function hasCustomEndpoint() {
    return getCustomApiUrl() && getCustomAppId();
}
```
Then we can use `getApiEndpoint` and `getDefaultEndpoint` to get all the data we need regarding the API endpoints.

This is how I would (personally) prefer to write this module, but of course it's not a complete solution (needs OAuth, language, etc.)

I'd put them all inside `endpoint.js`, and create another module (e.g. `account.js`) for validating the tokens and getting the client options.